ott/core/quad_problems.py

```diff
@@ -297,37 +296,35 @@ def cost_unbalanced_correction(
       marginal_2: jnp.ndarray<float>[num_b,], marginal of the transport matrix
         for samples from :attr:`geom_yy`.
       epsilon: regulariser.
-      rescale_factor: scaling factor for the transport matrix.
       delta: small quantity to avoid diverging KLs.
 
     Returns:
       The cost term.
     """
 
     def regulariser(tau: float) -> float:
-      return tau / (1.0 - tau) if tau != 1.0 else 0
+      return epsilon._target_init * tau / (1.0 - tau) if tau != 1.0 else 0
 
     cost = regulariser(self.tau_a) * jax.scipy.special.xlogy(
-        marginal_1, rescale_factor * marginal_1 / jnp.clip(self.a, a_min=delta)
+        marginal_1, marginal_1 / jnp.clip(self.a, a_min=delta)
```

Comment:
maybe split the log here as well, rather than have a division? this will then avoid having to clip, since if `a` has zero entries, `marginal_1` should also be 0.

Reply:
additionally, the same `marginal_1` / `self.a` `xlogy` evaluation is also provided below, maybe an opportunity to consolidate and make this slightly faster? (of course this will be negligible when compared to `xlogy` of transport.